### ISSUE TITLE
C++ SWIG Python interface Bug

### ISSUE THREAD

**Issue by attozeta** [swig/swig#1336]
Dear SWIG-Team,

there is a long-standing bug generating a Python wrapper for C++. See here: https://github.com/attozeta/swig-problem

Cheers,
attozeta

**Comment by wsfulton** [comment]
This  works:
```
python -c "import foo; f=foo.make_foo(); print(f.b.x)"
```
Have a look at the reference counting, I suspect the foo class instance is being destroyed before the inner bar class's x is read. http://swig.org/Doc3.0/Python.html#Python_nn19 may give some insight, but could be improved to warn about this.

**Comment by zeptomu** [comment]
Yes, the `foo` class is garbage collected and the workaround resolves this.
